astropy/utils/data.py

```diff
@@ -446,13 +446,13 @@ def get_pkg_data_fileobj(data_name, package=None, encoding=None, cache=True):
         all_urls = (conf.dataurl, conf.dataurl_mirror)
         for url in all_urls:
             try:
-                return get_readable_fileobj(url + datafn, encoding=encoding,
+                return get_readable_fileobj(url + data_name, encoding=encoding,
```

Comment:
@eteq - I'm surprised this ever worked before! I guess maybe we never used ``get_pkg_data_fileobj`` internally in the package before to get a remote file?

Reply:
Hmm, I *think* it was built assuming that the paths from `_find_pkg_data_path` are relative?  But that's clearly not the case now - not sure if it was an original bug or if that changed at some later date.